docs/reference/multi-tenancy.md

```diff
@@ -0,0 +1,383 @@
+
+## Custom Resources
+Ratify enables multi-tenancy support by allowing users to define resources in a namespace scope. This allows users to define policies, verifiers, stores, and key management providers in a namespace and have them apply only to that namespace. Additionally, users can define cluster-wide resources as defaults for namespaces that do not have a specific resource defined.
+
+Below is a list of Custom Resources that support multi-tenancy.
+| | Cluster-wide | Namespaced |
+|---|---|---|
+|[Referrer Store](./custom%20resources/stores.md)| Store | NamespacedStore |
+|[Verifier](./custom%20resources/verifiers.md)| Verifier | NamespacedVerifier |
+|[Policy](./custom%20resources/policies.md)| Policy | NamespacedPolicy |
+|[Key Management Provider](./custom%20resources/key-management-providers.md)| KeyManagementProvider | NamespacedKeyManagementProvider |
+
+**Note1**: CertificateStore resource is in **DEPRECATED** stage. If users want to move to multi-tenancy, they should use the Key Management Provider instead.
+
+**Note2**: In V2.0.0, we plan to rename cluster-wide resources to ClusterResource, e.g. ClusterVerifier, ClusterStore, ClusterPolicy and ClusterKeyManagementProvider, while renaming namespaced resources to Resource without prefix, e.g. Verifier, Store, Policy and KeyManagementProvider. This change will make the resource easier to understand and follow the conventions in Kubernetes community.
+
+## Multi-tenancy: Opt-In/Out
+
+There is no explicit feature flag to enable or disable multi-tenancy in Ratify. However, users would need to apply some resources to actively use multi-tenancy.
+
+### Custom Resources
+Users are supposed to use multi-tenancy by defining resources in a namespace scope or use cluster-wide resources as defaults. If users just want to enforce verification for the entire cluster, they should define resources in a cluster-wide scope.
+
+### Constraint Template
+Starting from version 1.2.0, Ratify now accommodates keys from external data request in both `[namespace]artifact-url` and `artifact-url` formats. To opt in multi-tenancy, users should adjust their constraint template to pass keys in `[namespace]artifact-url` format. However, if users just want to enforce verification for the entire cluster, they should pass only the `artifact-url` keys to Ratify.
+
+Examples:
+
+Constraint template using `artifact-url` formats: [default template](https://github.com/deislabs/ratify/blob/dev/library/default/template.yaml)
+
+Constraint template using `[namespace]artifact-url` formats: [multi-tenancy template](https://github.com/deislabs/ratify/blob/dev/library/multi-tenancy-validation/template.yaml)
+
+## Access Control
```

Comment:
Suggestion to move Access control section before Custom Resource. It is good to start with the concepts and the designed user groups in my opinion. :)